apps/server/lib/lexical/server/configuration/support.ex

```diff
@@ -39,20 +40,37 @@ defmodule Lexical.Server.Configuration.Support do
       |> get_in([:text_document, :signature_help])
       |> bool()
 
-    %__MODULE__{
+    work_done_progress? =
+      client_capabilities
+      |> get_in([:window, :work_done_progress])
+      |> bool()
+
+    config = %__MODULE__{
       code_action_dynamic_registration?: dynamic_registration?,
       hierarchical_document_symbols?: hierarchical_symbols?,
       snippet?: snippet?,
       deprecated?: deprecated?,
       tags?: tags?,
-      signature_help?: signature_help?
+      signature_help?: signature_help?,
+      work_done_progress?: work_done_progress?
     }
+
+    set(config)
+    config
   end
 
   def new(_) do
     %__MODULE__{}
   end
 
+  def get do
+    :persistent_term.get({__MODULE__, :config_support}, %__MODULE__{})
+  end
+
   defp bool(b) when b in [true, false], do: b
   defp bool(_), do: false
+
+  defp set(config) do
```

Comment:
shouldn't this be handled at the level of `Configuration` and not `Configuration.Support` ?